src/components/table-of-contents/table-of-contents.e2e.tsx

```diff
@@ -0,0 +1,136 @@
+import { newE2EPage } from '@stencil/core/testing';
+
+const sampleItems  = [
+  {
+    id: "item-one",
+    title: "Item One",
+  },
+  {
+    id: "item-two",
+    title: "Item Two"
+  },
+  {
+    id: "item-three",
+    title: "Item Three"
+  }
+];
+
+describe('la-table-of-contents', () => {
+  it('should render', async () => {
+    const page = await newE2EPage();
+    await page.setContent(`<la-table-of-contents></la-table-of-contents>`);
+    const component = await page.find('la-table-of-contents');
+    expect(component).not.toBeNull();
+  });
+
+  it('should render correct table of contents html for the given set items prop', async () => {
+    const page = await newE2EPage();
+    await page.setContent(`<la-table-of-contents items='${JSON.stringify(sampleItems)}'></la-table-of-contents>`);
+    const renderedItems = await page.findAll('la-table-of-contents la-toc-item .content__action__title');
+
+    const dataMatchesDOMNodes = (
+      items: any[],
+      data: Array<{
+        id: string;
+        title: string;
+      }>,
+    ) => {
+      return items.every((item: { innerText: string; href: string }, index: number) => {
+        return item.innerText === data[index].title;
+      });
+    };
+
+    expect(dataMatchesDOMNodes(renderedItems, sampleItems)).toBe(true);
+  });
+
+  it('should filter on mount', async () => {
+    const page = await newE2EPage();
+    await page.setContent(
+      `<la-table-of-contents items='${JSON.stringify(sampleItems)}' title-filter='Item One'></la-table-of-contents>`
+    );
+
+    const renderedItems = await page.findAll('la-table-of-contents la-toc-item');
+    expect(renderedItems[0].className.includes('excluded')).toBe(false);
+    expect(renderedItems[1].className.includes('excluded')).toBe(true);
+    expect(renderedItems[2].className.includes('excluded')).toBe(true);
+  });
+
+  it('should filter items on change of titleFilter prop', async () => {
```

Comment:
Good edge cases to include when doing tests on string inputs are:

* empty string
* string that is just spaces
* leading and trailing spaces
* case sensitivity

Reply:
In terms of behaviour, I propose: search is case insensitive (I think it is already) and leading and trailing spaces are stripped from inputs before they're used.

Reply:
added test cases for the above